src/find_bottleneck.rs

```diff
@@ -199,24 +230,39 @@ impl FindBottleneck {
         }
         Ok(())
     }
+
     /// Writes the asn bottleneck result to a stdout or a time stamped file
-    pub(crate) fn write(self, out: Option<&Path>) -> Result<()> {
-        if let Some(path) = out {
-            let epoch = SystemTime::now()
-                .duration_since(SystemTime::UNIX_EPOCH)
-                .unwrap();
-            let now = epoch.as_secs();
-            let dst = path.join(format!("bottleneck.{}.txt", now));
-            let mut file = File::create(&dst).map_err(|io_error| Error::IoError {
-                io_error,
-                path: dst.to_path_buf(),
-            })?;
+    fn write_intermediate(self, end_high_octet: u8) -> Result<()> {
+        let file_name = format!("tmp/bottleneck.up_to_{}.txt", end_high_octet);
+        let dst = PathBuf::from(file_name);
+        let mut file = File::create(&dst).map_err(|io_error| Error::IoError {
+            io_error,
+            path: dst.to_path_buf(),
+        })?;
+        self.write_bottleneck(&mut file)?;
+        Ok(())
+    }
 
-            self.write_bottleneck(&mut file)?;
+    /// Writes the asn bottleneck result to a stdout or a time stamped file
+    pub(crate) fn write_result(out: Option<&Path>) -> Result<()> {
+        let epoch = SystemTime::now()
+            .duration_since(SystemTime::UNIX_EPOCH)
+            .unwrap();
+        let now = epoch.as_secs();
+        let file_name = format!("bottleneck.{}.txt", now);
+        let dst;
+        if let Some(path) = out {
+            dst = path.join(file_name);
         } else {
-            self.write_bottleneck(&mut io::stdout())?;
```

Comment:
Why are we removing the stdout print option?

Reply:
This is because I write result in parts, and there will be logging in between parts (to track progress). I decided it's better to default to a file, rather than to stdout.

Reply:
Ok, that is fine. We just need to fix different parts of the code then. The subcommand, many comments throughout the code, and the README.